### ISSUE TITLE
accuracy reduce during the training

### ISSUE THREAD

**Issue by Sxx1995** [jadore801120/attention-is-all-you-need-pytorch#60]
Hi. I just follow the tutorial to train the model with the dataset given here. However, the accuracy is relatively high at epoch 0 and has a sharp decline after that. Does anybody meet the similar problem? 

Here is the record:

[ Epoch 0 ]
  - (Training)   ppl:  69.89619, accuracy: 48.127 %, elapse: 9.800 min
  - (Validation) ppl:  86.35283, accuracy: 20.530 %, elapse: 3.096 min
    - [Info] The checkpoint file has been updated.
[ Epoch 1 ]
  - (Training)   ppl:  135.71178, accuracy: 32.377 %, elapse: 10.807 min
  - (Validation) ppl:  865.33501, accuracy: 5.777 %, elapse: 3.052 min
[ Epoch 2 ]
  - (Training)   ppl:  193.38618, accuracy: 27.988 %, elapse: 11.013 min
  - (Validation) ppl:  949.73713, accuracy: 4.359 %, elapse: 3.093 min

**Comment by dksehdals216** [comment]
I can confirm this and has sent @jadore801120  an email previously. 
This seems to be the general case after the recent commits a few days ago. I have also confirmed on the WMT '14 and '13.

**Comment by ivan-bilan** [comment]
I also get lower performance overall with the new 0.4.1 update. I am only using the Encoder part in my project, however.

**Comment by ivan-bilan** [comment]
I also get lower performance overall with the new 0.4.1 update. I am only using the Encoder part in my project, however.

**Comment by jadore801120** [comment]
Hello @ivan-bilan , thanks a lot for point out! 
So the bug may exist in encoder part.
I will take a closer look.

**Comment by limteng-rpi** [comment]
Hi @jadore801120 , have you identified the causes of this accuracy issue? Thanks.

**Comment by Belerafon** [comment]
I suppose that learning rate is very high and accuracy goes down after warming up steps. And success depends on seed or something. Try to set smaller lr.

**Comment by smukh93** [comment]
I set lr to 0 but haven't got any improvements. Anyone else found a solution?

**Comment by JuliRao** [comment]
Hi @jadore801120 Have you fixed the problem? thanks. I'm running my task on your the modification of your model. When number of layer is set to 1, the result is okay. However, when the number of layers is set to 2, the performance drops quickly.
